scripts/portfolio.js

```diff
@@ -1,108 +1,103 @@
-
-  dataPassedIn.sort(function(a, b) {
-    console.log('here');
-    return (new Date(b.launchedOn)) - (new Date(a.launchedOn));
-  });
-
-  dataPassedIn.forEach(function(ele) {
-    Project.all.push(new Project(ele));
-  });
-
-  // projects.forEach(function(a){
-  //   $('#projects').append(a.toHtml());
-  // });
-};
-  // $('#author-filter').append(a.toHtml('author'));
-  // if (categories.indexOf(a.category) === -1) {
-  //   $('#category-filter').append(a.toHtml('category'));
-  //   categories.push(a.category);
-  // }
-// });
-
-// projects.forEach(function(a){
-//   $('#author-filter').append(a.toAuthors());
-// });
-//
-// projects.forEach(function(a){
-//   $('#category-filter').append(a.toCategories());
-// });
-Project.fetchAll = function() {
-  if (localStorage.projects) {
-    $.ajax({
-      type: 'HEAD',
-      url: 'data/projects.json',
-      success: function(data, message, xhr) {
-        var etag = (xhr.getResponseHeader('ETag'));
-        if (localStorage.etag === etag) {
-          Project.loadAll(JSON.parse(localStorage.projects));
-          projectView.initIndexPage();
-        } else {
-          Project.jsonfetch();
+(function(module) {
+
+  function Project (opts) {
+    this.title = opts.title;
+    this.author = opts.author;
+    this.category = opts.category;
+    this.launchedOn = opts.launchedOn;
+    this.projectLink = opts.projectLink;
+    this.body = opts.body;
+  };
+
+  Project.all = [];
+
+  Project.prototype.toHtml = function(){
+    // var $source = $('#' + templateId + '-template').html();
+    var template = Handlebars.compile($('#projects-template').text());
+
+    this.daysAgo = parseInt((new Date() - new Date(this.launchedOn))/60/60/24/1000);
+    this.publishStatus = this.launchedOn ? 'launched ' + this.daysAgo + ' days ago' : '(draft)';
+    return template(this);
+  };
+
+  Project.loadAll = function(dataPassedIn) {
+
+    dataPassedIn.sort(function(a, b) {
+      console.log('here');
+      return (new Date(b.launchedOn)) - (new Date(a.launchedOn));
+    });
+
+    Project.all = dataPassedIn.map(function(ele) {
+      return new Project(ele);
+    });
+  };
+
+  Project.fetchAll = function(callback) {
+    if (localStorage.projects) {
+      $.ajax({
+        type: 'HEAD',
+        url: '/data/projects.json',
+        success: function(data, message, xhr) {
+          var eTag = xhr.getResponseHeader('ETag');
+          if(!localStorage.eTag || eTag !== localStorage.eTag) {
+            localStorage.eTag = eTag;
+            Project.getAll();
+          } else {
+            Project.loadAll(JSON.parse(localStorage.projects));
+            callback();
+          }
         }
-      }
+      });
+    } else {
+      Project.getAll();
+    }
+  };
+
+  Project.getAll = function() {
+    $.getJSON('/data/projects.json', function(responseData) {
+      Project.loadAll(responseData);
+      localStorage.projects = JSON.stringify(responseData);
+      projectView.initIndexPage();
     });
```

Comment:
You should refactor this so it takes a callback instead of hardcoding in the projectview.initIndexPage() function.